src/parser.rs

```diff
@@ -373,7 +377,11 @@ impl Parser {
             Token::LCurly => self.record_or_closure(),
             Token::LParen => {
                 self.tokens.advance();
-                let output = self.expression();
+                let output = if self.tokens.peek().0 == Token::RParen {
+                    self.error("use null instead of ()")
+                } else {
+                    self.expression()
+                };
                 self.rparen();
```

Comment:
This one can be moved into the `else` branch with `self.expression()` to prevent the redundant `expected: right paren ')'` error seen in the last match.nu test output.